test/test_utils.dart

```diff
@@ -1,6 +1,31 @@
+import 'dart:async';
+import 'dart:io';
+
 import 'package:flutter/material.dart';
 
-int generateTestingPortFromName(String name) => 50000 + (name.hashCode % 10000);
+Future<int> getUnusedPort() async {
+  int? value;
+  await Future.doWhile(() async {
+    value = await ServerSocket.bind(InternetAddress.loopbackIPv4, await getUnsafeUnusedPort()).then((socket) {
+      final port = socket.port;
+      socket.close();
+      return port;
+    });
```

Comment:
Do we need to test the bind here, since the socket binding is already being tested in `getUnsafeUnusedPort`?

Reply:
Yeah, good question. I think I was misunderstanding the code I cribbed from, it adds this additional step but returns a `T` instead, the goal is to avoid a race condition that occurs when some other process binds to the port returned by `getUnsafeUnusedPort` after it returns. So basically repeatedly grab a _probably_ unused port then attempt to actually do something with it and then when you are able to do so successfully, return it. 

But since the issue we're running into is an error being thrown rather than a null return, this won't actually work smoothly without a bit more work. Will revisit and re-request review.